src/include/duckdb/main/client_context.hpp

```diff
@@ -325,4 +326,32 @@ class ClientContextLock {
 	lock_guard<mutex> client_guard;
 };
 
+//! The QueryContext wraps an optional client context.
+//! It makes query-related information available to operations.
+class QueryContext {
+public:
+	QueryContext() : context(nullptr) {
+	}
+	QueryContext(ClientContext &context) : context(context) {
+	}
+	QueryContext(optional_ptr<ClientContext> context) : context(context) {
```

Comment:
This constructor should not be necessary - `optional_ptr<ClientContext>` is already implicitly constructable from `ClientContext &`